Makefile

```diff
@@ -35,36 +36,66 @@ C_BINDING_CLEAN = $(foreach dir,$(C_BINDING_DIRS),$(dir)Makefile $(dir).qmake.st
 # apply formatting / style guidelines
 lint: fmt_check clippy
 
-# idempotent install rustup with the default toolchain set for tooling
-# best for CI based on tools only
-.PHONY: install_rustup
-install_rustup:
-	if ! which rustup ; then \
-		curl https://sh.rustup.rs -sSf | sh -s -- --default-toolchain $(CORE_RUST_VERSION) -y; \
+# Check if Rust version is correct, and prompts to offer to change to the correct version.  Requires
+# RUST_VERSION to be set, as appropriate for whatever target is being installed (defaults to
+# CORE_RUST_VERSION; see install_rustup..., below).  We'll also export PATH to default location of
+# Rust installation for use here in the Makefile, in case this is the first time rustup has been
+# installed/run, and we don't have a rustup-modified .profile loaded yet.  If not connected to a
+# terminal (stdin is a tty), or running under a Continuous Integration test (CI), defaults to
+# automatically installing and changing the default Rust version (under the assumption that the
+# invoker of the Makefile target knows what they want, under headless automated procedures like
+# CI). Otherwise, entering "no<return>" rejects installing/changing the Rust version (and we assume
+# you know what you're doing, eg. testing some new Rust toolchain version that you've installed)
+export PATH := $(HOME)/.cargo/bin:$(PATH)
+RUST_VERSION = $(CORE_RUST_VERSION)
+.PHONY: version_rustup
+version_rustup:
+	@if which rustup; then \
+	    echo "\033[0;93m## Current Rust version installed (need: '$(RUST_VERSION)'): ##\033[0m"; \
+	    if ! rustup show 2>/dev/null | grep -qe "$(RUST_VERSION).*(default)"; then \
+	        rustup show; \
+		echo "\033[0;93m## Change current Rust version to '$(RUST_VERSION)' ##\033[0m"; \
+	        [ -t 1 ] && [[ "$(CI)" == "" ]] && read -p "Continue? (Y/n) " yes; \
+	        if [[ "$${yes:0:1}" != "n" ]] && [[ "$${yes:0:1}" != "N" ]]; then \
+	            echo "\033[0;93m## Selecting Rust version '$(RUST_VERSION)'... ##\033[0m"; \
+	            rustup default $(RUST_VERSION); \
```

Comment:
`rustup` has a override feature that should be used here instead of `rustup default`

Reply:
For example:
```shell
rustup override set $(RUST_VERSION)
```
will set the the rustc and cargo version only for the current directory

Reply:
Ah; so there are *two* places that need to be checked, in order, to determine the version of toolchain selected; rustup override list (to determine if the correct toolchain is already selected), and then rustup show (to see if A) if the toolchain is installed, and B) if it is the default).

Actually, it looks like it'll automatically install the correct toolchain if it doesn't exist, so only `rustup override ...` is required to test/set toolchain selection.  OK, I'll make the change.